hail/python/hailtop/aiotools/weighted_semaphore.py

```diff
@@ -23,16 +24,18 @@ class WeightedSemaphore:
     def __init__(self, value: int):
         self.max = value
         self.value = value
-        self.events: List[Tuple[int, asyncio.Event]] = []
+        self.events: SortedKeyList[Tuple[int, asyncio.Event]] = SortedKeyList(key=lambda x: x[0])
 
     def release(self, n: int) -> None:
         self.value += n
         while self.events:
-            n, event = self.events[-1]
-            if n >= self.value:
-                self.events.pop()
+            n, event = self.events[0]
```

Comment:
We could also do a bisect_left to get the largest n possible that is less than or equal to self.value.

Reply:
Again, possible future optimization.